k8s/upgrade-job/src/helm/values.rs

```diff
@@ -107,7 +108,14 @@ pub(crate) fn generate_values_args(
 
     upgrade_args.push("--set".to_string());
 
-    let image_tag_arg = format!("{image_key}.{tag_key}={}", to_values.image_tag());
+    let image_tag_arg = match custom_image_tag {
+        Some(image_tag) => {
+            format!("{image_key}.{tag_key}={}", image_tag.as_str())
+        }
+        None => {
+            format!("{image_key}.{tag_key}={}", to_values.image_tag())
+        }
+    };
```

Comment:
```suggestion
   let image_tag_arg = format!(
    "{image_key}.{tag_key}={}",
    custom_image_tag.unwrap_or_else(|| to_values.image_tag().as_str())
);
```
Would this be simple?